jwst/outlier_detection/_fileio.py

```diff
@@ -20,13 +12,46 @@ def save_median(median_model, make_output_path, asn_id=None):
     median_model : ~jwst.datamodels.ImageModel
         The median ImageModel or CubeModel to save
     '''
-    default_suffix = "_outlier_i2d.fits"
-    if asn_id is None:
-        suffix_to_remove = default_suffix
-    else:
-        suffix_to_remove = f"_{asn_id}{default_suffix}"
-    median_model_output_path = make_output_path(
-        basepath=median_model.meta.filename.replace(suffix_to_remove, '.fits'),
-        suffix='median')
-    median_model.save(median_model_output_path)
-    log.info(f"Saved model in {median_model_output_path}")
+    _save_intermediate_output(median_model, "median", make_output_path)
+
+
+def save_drizzled(drizzled_model, make_output_path):
+    expected_tail = "outlier_?2d.fits"
+    suffix = drizzled_model.meta.filename[-len(expected_tail):-5]
+    _save_intermediate_output(drizzled_model, suffix, make_output_path)
+
+
+def save_blot(input_model, blot, make_output_path):
+    blot_model = _make_blot_model(input_model, blot)
+    _save_intermediate_output(blot_model, "blot", make_output_path)
+
+
+def _make_blot_model(input_model, blot):
+    blot_model = type(input_model)()
+    blot_model.data = blot
+    blot_model.update(input_model)
+    return blot_model
+
+
+def _save_intermediate_output(model, suffix, make_output_path):
+    """
+    Ensure all intermediate outputs from OutlierDetectionStep have consistent file naming conventions
+    
+    Notes
+    -----
+    self.make_output_path() is updated globally for the step in the main pipeline
+    to include the asn_id in the output path, so no need to handle it here.
+    """
+
+    # make_output_path cannot handle suffix with an underscore inside it,
```

Comment:
`make_output_path` can handle suffixes with underscores. `flat_field` is one example:
https://github.com/spacetelescope/jwst/blob/main/jwst/lib/suffix.py#L121
The issue is that `outlier_i2d` is not a known suffix:
```python
>>> import jwst.lib.suffix
>>> 'outlier_s2d' in jwst.lib.suffix.KNOW_SUFFIXES
False
```
I'd say let's not try to address that in this PR but suggest that this comment be removed or updated.

Reply:
updated the comment, thanks for pointing this out.  I was not aware that KNOW_SUFFIXES was checked

Reply:
It's round-about with `Step.make_output_path` calling `Step.remove_suffix` that calls https://github.com/spacetelescope/jwst/blob/e8ba844c92f5959069859aea0336214360f2d75d/jwst/lib/suffix.py#L208
that uses a regex made from `KNOW_SUFFIXES`:
https://github.com/spacetelescope/jwst/blob/e8ba844c92f5959069859aea0336214360f2d75d/jwst/lib/suffix.py#L327

Reply:
gotcha, thanks